recipes-app/node-red-gpio/node-red-gpio_0.0.6-IOT2050.bb

```diff
@@ -20,8 +20,7 @@ SRC_URI = " \
     file://0002-extend-gpio-to-D19.patch \
     file://0003-Clean-up-mraa-objects-on-node-closing.patch \
     file://0004-Add-USER-button.patch \
-    file://0005-Add-configurable-GPIO-mode-for-DIN.patch \
-    file://0006-node-red-gpio-Add-support-for-initial-message-from.patch"
+    file://0005-Add-configurable-GPIO-mode-for-DIN.patch"
```

Comment:
Can you provide an explanation why the pattern no longer works here? Did something in Node Red change since the version that the IOT2000 is using (where this definitely worked)?

Reply:
First of all - I didn't test it well, I've looked through physical behavior of this thing which is working except that it didn't send anything on startup from node-red pov, it just calls the handler which just send on GPIO behavior change from falling to raising and from raising to falling. So, I've missed that point with node.send.

Reply:
> Ok, but then you do not need to add an extra trigger input to the DIN node, only fix the "send once on startup" path.

I thought it first too but then you have to control some needs, like sending status over time or on startup and this possible to make it with inject node which provides exact this stuff related to timings without any enhancements to GPIO and it still be possible to use without input.

Reply:
You only need the additional on-startup trigger, and that could be internal. No need for a timer, all inputs should have edge triggers which will send out new messages automatically.

Reply:
> You only need the additional on-startup trigger, and that could be internal. No need for a timer, all inputs should have edge triggers which will send out new messages automatically.

But it the same as for inject nodes which can provide such functionality and even bigger, you can control status reporting via this node which provide you more than just trigger on startup.

Reply:
I'd still like to see some concrete GPIO node which does the same. I would not try to attach an input to something that conceptually only delivers outputs.

Reply:
> I'd still like to see some concrete GPIO node which does the same. I would not try to attach an input to something that conceptually only delivers outputs.

Ok, I'll just fix that port then with sending out on startup but what with proposition about cycle-timers from #315 ?

Reply:
Why do we need a timer? Is any of the input not sending events on falling or rising edges?

Reply:
> Why do we need a timer?

As example, if client wants to report about status somewhere else by timer or event at startup both of these things could be provided by inject node. So, it's like role distribution between nodes, inject nodes provides any time triggers(interval, once on startup, etc.), gpio node provides edge triggers and doesn't have any other internal logic about time, inject node provides that stuff.

> Is any of the input not sending events on falling or rising edges?

Yes, it should send on edge event.